--- a/bin/daps.in
+++ b/bin/daps.in
@@ -1732,7 +1732,7 @@
     if [[ ${DOCBOOK5_STYLE_URI:0:5} = file: ]]; then
         DOCBOOK_STYLES="$DOCBOOK5_STYLE_URI"
     else
-        DOCBOOK_STYLES=$(xmlcatalog "$XML_MAIN_CATALOG" \
+        DOCBOOK_STYLES=$(xmlcatalog "" \
           "$DOCBOOK5_STYLE_URI" 2>/dev/null) || \
             exit_on_error "Could not determine the DocBook stylesheet location by resolving \"$DOCBOOK5_STYLE_URI\" via xmlcatalog"
     fi
@@ -1748,13 +1748,13 @@
             # leaves loop via break, otherwise continue with 5.0
             #
             D5U="${DOCBOOK5_RNG_URI/@db5version\@/$DB5_VERSION}"
-            DOCBOOK5_RNG=$(xmlcatalog "$XML_MAIN_CATALOG" "$D5U") && break
+            DOCBOOK5_RNG=$(xmlcatalog "" "$D5U") && break
         done
     else
         if [[ ${DOCBOOK5_RNG_URI:0:5} = file: ]]; then
             DOCBOOK5_RNG="$DOCBOOK5_RNG_URI"
         else
-            DOCBOOK5_RNG=$(xmlcatalog "$XML_MAIN_CATALOG" \
+            DOCBOOK5_RNG=$(xmlcatalog "" \
               "$DOCBOOK5_RNG_URI" 2>/dev/null) || \
                 exit_on_error "Could not determine the DocBook 5 schema location by resolving \"$DOCBOOK5_RNG_URI\" via xmlcatalog"
         fi
@@ -1782,7 +1782,7 @@
     if [[ ${DOCBOOK4_STYLE_URI:0:5} = file: ]]; then
         DOCBOOK_STYLES="$DOCBOOK4_STYLE_URI"
     else
-        DOCBOOK_STYLES=$(xmlcatalog "$XML_MAIN_CATALOG" \
+        DOCBOOK_STYLES=$(xmlcatalog "" \
           "$DOCBOOK4_STYLE_URI" 2>/dev/null) || \
             exit_on_error "Could not determine the DocBook stylesheet location by resolving \"$DOCBOOK4_STYLE_URI\" via xmlcatalog"
     fi
--- a/configure.ac
+++ b/configure.ac
@@ -159,12 +159,6 @@
 AC_PATH_PROG([JING], [jing], [0])
 AC_PATH_PROG([TRANG], [trang], [0])
 
-
-AC_CHECK_FILE([$root_catalog], [CATALOG=$root_catalog], [CATALOG=0])
-if test 0 = "$CATALOG"; then
-  AC_MSG_ERROR([Catalog $root_catalog not found])
-fi
-
 dnl -------------------------------------------------------------------------
 AC_MSG_NOTICE([===== Checking for Image conversion tools...])
 
@@ -199,7 +193,7 @@
 for SVG_DTD in "-//W3C//DTD SVG 1.0//EN" \
                "-//W3C//DTD SVG 1.1 Basic//EN" ; do
   AC_MSG_CHECKING([for $SVG_DTD])
-  if AC_RUN_LOG([$XMLCATALOG --noout "$root_catalog" "$SVG_DTD" >&2]); then
+  if AC_RUN_LOG([$XMLCATALOG --noout "" "$SVG_DTD" >&2]); then
     AC_MSG_RESULT([yes])
     HAVE_SVGDTD=1
   else
@@ -235,7 +229,7 @@
                "-//OASIS//DTD DocBook XML V4.4//EN" \
                "-//OASIS//DTD DocBook XML V4.5//EN" ; do
   AC_MSG_CHECKING([for $XML_DTD])
-  if AC_RUN_LOG([$XMLCATALOG --noout "$root_catalog" "$XML_DTD" >&2]); then
+  if AC_RUN_LOG([$XMLCATALOG --noout "" "$XML_DTD" >&2]); then
     AC_MSG_RESULT([yes])
     HAVE_DOCBOOK=1
   else
@@ -258,7 +252,7 @@
              "-//OASIS//DTD XML Exchange Table Model 19990315//EN" \
              ; do
  AC_MSG_CHECKING([for $IDENT])
- if AC_RUN_LOG([$XMLCATALOG --noout "$root_catalog" "$IDENT" >&2]); then
+ if AC_RUN_LOG([$XMLCATALOG --noout "" "$IDENT" >&2]); then
     AC_MSG_RESULT([yes])
     HAVE_DOCBOOK_IDENTIFIER=1
   else
@@ -293,7 +287,7 @@
            "ISO 8879:1986//ENTITIES Non-Russian Cyrillic//EN//XML" \
              ; do
  AC_MSG_CHECKING([for $ENT])
- if AC_RUN_LOG([$XMLCATALOG --noout "$root_catalog" "$ENT" >&2]); then
+ if AC_RUN_LOG([$XMLCATALOG --noout "" "$ENT" >&2]); then
     AC_MSG_RESULT([yes])
     HAVE_ISO_ENTITIES=1
   else
@@ -313,7 +307,7 @@
                   "http://docbook.org/xml/5.0/rng/docbook.rng" \
                   "http://docbook.org/xml/5.0/rng/docbook.rnc" ; do
   AC_MSG_CHECKING([for $XML_SCHEMA])
-  if AC_RUN_LOG([$XMLCATALOG --noout "$root_catalog" "$XML_SCHEMA" >&2]); then
+  if AC_RUN_LOG([$XMLCATALOG --noout "" "$XML_SCHEMA" >&2]); then
     AC_MSG_RESULT([yes])
     HAVE_DOCBOOK_50=1
     db5_version="5.0"
@@ -337,7 +331,7 @@
                   "http://docbook.org/xml/5.1/rng/dbits.rng" \
                   "http://docbook.org/xml/5.1/rng/dbits.rnc"  ; do
   AC_MSG_CHECKING([for $XML_SCHEMA])
-  if AC_RUN_LOG([$XMLCATALOG --noout "$root_catalog" "$XML_SCHEMA" >&2]); then
+  if AC_RUN_LOG([$XMLCATALOG --noout "" "$XML_SCHEMA" >&2]); then
     AC_MSG_RESULT([yes])
     HAVE_DOCBOOK_51=1
     db5_version="5.1"
@@ -374,7 +368,7 @@
             "${DB4_XSLT_URL}xhtml/chunk.xsl" \
             "${DB4_XSLT_URL}xhtml/docbook.xsl"; do
   AC_MSG_CHECKING([for $XSLT])
-  if AC_RUN_LOG([$XMLCATALOG --noout "$root_catalog" "$XSLT" >&2]); then
+  if AC_RUN_LOG([$XMLCATALOG --noout "" "$XSLT" >&2]); then
     AC_MSG_RESULT([yes])
     HAVE_DOCBOOK_STYLESHEETS=1
   else
@@ -397,7 +391,7 @@
             "${DB5_XSLT_URL}xhtml/chunk.xsl" \
             "${DB5_XSLT_URL}xhtml/docbook.xsl"; do
   AC_MSG_CHECKING([for $XSLT])
-  if AC_RUN_LOG([$XMLCATALOG --noout "$root_catalog" "$XSLT" >&2]); then
+  if AC_RUN_LOG([$XMLCATALOG --noout "" "$XSLT" >&2]); then
     AC_MSG_RESULT([yes])
     HAVE_DOCBOOK_5_STYLESHEETS=1
   else
